src/app/core/nav-menu/nav-menu.service.spec.ts

```diff
@@ -0,0 +1,45 @@
+import { TestBed } from '@angular/core/testing';
+import { NavMenuService } from './nav-menu.service';
+import { Store } from '@ngrx/store';
+import { TestStore } from '../../../testutils/ngrx-test-store';
+import { appNavRoutes } from './routes';
+
+describe('NavMenuService', () => {
+  let service: NavMenuService;
+  let store: TestStore<string>;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        NavMenuService,
+        { provide: Store, useClass: TestStore }
+      ]
+    });
+    service = TestBed.get(NavMenuService);
+    store = TestBed.get(Store);
+  });
+
+  it('should return an array of routes', () => {
```

Comment:
As far as I know, tests with async tasks inside should be wrapped into `async` or you have to use `done` param. Seems like it is working only because of the nature of `Observable.of`, actually, it works synchronously. Just try to put a delay after the `Observable.of` and you will get trouble.